### PR TITLE
Refactoring sender/receiver stuff - post-merge

### PR DESCRIPTION
**Pull request by khs26**
Supercedes #428

### CODE REVIEW

File: src/common/pty.rs [hunk 1]
```diff
@@ -124,9 +176,7 @@ fn stream_terminal_bytes(
                         Some(receive_time) => {
                             if receive_time.elapsed() > max_render_pause {
                                 pending_render = false;
-                                send_screen_instructions
-                                    .send(ScreenInstruction::Render)
-                                    .unwrap();
+                                let _ = send_screen_instructions.send(ScreenInstruction::Render);
```
Comment (khs26) [inline src/common/pty.rs:179]: Update this

File: src/common/pty.rs [hunk 2]
```diff
@@ -4,20 +4,19 @@ use ::async_std::task::*;
...
-use crate::layout::Layout;
+use super::{screen::ScreenInstruction, thread_bus::SenderWithContext};
 use crate::os_input_output::ServerOsApi;
 use crate::utils::logging::debug_to_file;
+use crate::wasm_vm::PluginInstruction;
 use crate::{
-    errors::{get_current_ctx, ContextType, ErrorContext},
+    common::thread_bus::Bus,
+    errors::{get_current_ctx, ContextType, PtyContext},
     panes::PaneId,
```
Comment (khs26) [inline src/common/pty.rs:17]: Choose one or the other

File: src/common/screen.rs [hunk 3]
```diff
@@ -3,15 +3,14 @@
...
+use crate::common::pty::{PtyInstruction, VteBytes};
+use crate::common::thread_bus::Bus;
+use crate::errors::{ContextType, ScreenContext};
 use crate::panes::PositionAndSize;
-use crate::pty_bus::{PtyInstruction, VteBytes};
 use crate::server::ServerInstruction;
 use crate::tab::Tab;
-use crate::{errors::ErrorContext, wasm_vm::PluginInstruction};
+use crate::wasm_vm::PluginInstruction;
 use crate::{layout::Layout, panes::PaneId};
```
Comment (khs26) [inline src/common/screen.rs:14]: And again

File: src/common/wasm_vm.rs [hunk 4]
```diff
@@ -1,17 +1,28 @@
...
-use wasmer_wasi::WasiEnv;
+use wasmer::{
+    imports, ChainableNamedResolver, Function, ImportObject, Instance, Module, Store, Value,
+    WasmerEnv,
+};
+use wasmer_wasi::{Pipe, WasiEnv, WasiState};
 use zellij_tile::data::{Event, EventType, PluginIds};
 
-use super::{pty_bus::PtyInstruction, screen::ScreenInstruction, PaneId, SenderWithContext};
+use super::{
```
Comment (khs26) [inline src/common/wasm_vm.rs:19]: :(

File: src/server/mod.rs [hunk 5]
```diff
@@ -224,588 +176,69 @@ fn init_session(
...
-            let send_screen_instructions = send_screen_instructions.clone();
-            let send_pty_instructions = send_pty_instructions.clone();
-            let send_plugin_instructions = send_plugin_instructions.clone();
-
+            let plugin_bus = Bus::new(
+                plugin_receiver,
+                Some(&to_screen),
+                Some(&to_pty),
+                Some(&to_plugin),
+                Some(&to_server),
```
Comment (khs26) [inline src/server/mod.rs:225]: do we send stuff to the server?

File: src/server/route.rs [hunk 6]
```diff
@@ -0,0 +1,205 @@
+use std::sync::{Arc, RwLock};
+
+use zellij_tile::data::Event;
+
+use crate::common::errors::{ContextType, ServerContext};
+use crate::common::input::actions::Action;
+use crate::common::input::{actions::Direction, handler::get_mode_info};
```
Comment (khs26) [inline src/server/route.rs:7]: :(

File: src/server/route.rs [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            session
+                .senders
+                .send_to_screen(ScreenInstruction::UpdateTabName(c))
+                .unwrap();
+        }
+        Action::NoOp => {}
+        Action::Quit => panic!("Received unexpected action"),
+    }
+}
+pub fn route_thread_main(
```
Comment (khs26) [inline src/server/route.rs:171]: newline

**Review by a-kenji — COMMENTED** [review]
This was still pending!

**Pull request merged**